tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,274 @@
+
+xdata = np.arange(0, 2 * np.pi, 0.01)
+(line,) = ax.plot(xdata, np.sin(xdata), c="b")
+ax.grid()
+ax.set_ylim(-1.1, 1.1)
+
+
+def update(frame):
+    # .set_ydata resets the y-data for the line, so we add the new point to
+    # the existing line x-data and calculate y again.
+    line.set_ydata(np.sin(xdata + frame / 50))
+    return (line,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update,
+                              interval=30)
+plt.show()
+
+###############################################################################
+# Animating Markers
+# ^^^^^^^^^^^^^^^^^
+#
+# `.pyplot.scatter` returns a :class:`~matplotlib.collections.PathCollection`
+# that can similarly be modified by using the
+# `.collections.PathCollection.set_offsets` function.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+t = np.linspace(-4, 4, 400)
+a, b = 3, 2
+delta = np.pi / 2
+
+scat = ax.scatter(np.sin(a * t[0] + delta), np.sin(b * t[0]), c="b", s=2)
+ax.grid()
+ax.set_xlim(-1.5, 1.5)
+ax.set_ylim(-1.5, 1.5)
+
+
+def update(frame):
+    # .set_offsets also resets the offset data for the entire collection with
+    # the new values. Therefore, to also carry forward the previously
+    # calculated information, we use the data from the first to the current
+    # frame to set the new offsets.
+    x = np.sin(a * t[:frame] + delta)
+    y = np.sin(b * t[:frame])
+    data = np.stack([x, y]).T
+    scat.set_offsets(data)
+    return (scat,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=400, interval=30)
+plt.show()
+
+
+###############################################################################
+# Animating Images
+# ^^^^^^^^^^^^^^^^
+#
+# When we plot an image using `.pyplot.imshow`, it returns an
+# :class:`~matplotlib.image.AxesImage` object. The data in this object can also
+# similarly be modified by using the `.image.AxesImage.set_data` method.
+
+
+def f(x, y, mean, cov):
+    dev_x = x - mean
+    dev_y = y - mean
+    maha = -0.5 * (((x-mean)/cov)**2 + ((y-mean)/cov)**2)
+    return (1/(np.pi * cov)) * np.exp(maha)
+
+fig, ax = plt.subplots()
+
+x, y = np.meshgrid(np.arange(-1, 1, 0.01), np.arange(-1, 1, 0.01))
+mean = 0
+cov = 0.1
+data = f(x, y, mean, cov)
+aximg = ax.imshow(data)
+
+
+def update(frame):
+    x, y = np.meshgrid(np.arange(-1, 1, 0.01), np.arange(-1, 1, 0.01))
+    mean = 0
+    cov = 0.01 * frame + 1e-6
+    data = f(x, y, mean, cov)
+
+    aximg.set_data(data)
+    return (aximg,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=None, interval=100)
+plt.show()
+
+###############################################################################
+# :class:`~matplotlib.animation.ArtistAnimation`
+# ----------------------------------------------
+#
+# On the other hand, :class:`~matplotlib.animation.ArtistAnimation` can be used
+# to generate animations if we have data on various different artists. This
+# list of artists is then converted frame by frame into an animation.
```

Comment:
To be honest, I don't know a good example where ArtistAnimation is useful. Maybe somebody else has an idea?

This example looks like `FuncAnimation` and useing `AxesImage.set_data` is simpler and much more memory efficient.

Reply:
I think something like `streamplot` or `contour` where updating the Artists is hard.  You do pay the memory cost, but it is possible!

Reply:
I agree that the example that I've shown is not something that would be done because using `FuncAnimation` is a better choice for that. One of the reasons I chose that was to show how both methods can achieve the same end goal. I do agree with the suggestion to use something that is difficult to update/doesn't have a straightforward `set_*` method. I'll look into the docs to come up with such an example.

Reply:
Bar charts seem to be a good option. Since it returns a container object, if we want to use `FuncAnimation`, we'd have to go and individually set the artists for each of them but with `ArtistAnimation`, we can do multiple plots. I implemented a small example. Does that look better than the one we have right now?